src/component/chat.cpp

```diff
@@ -101,6 +105,67 @@ namespace chat
 				g_say_hook.invoke<void>(ent, target, mode, chatText);
 			}
 		}
+
+		
+		// exemple:
+		// { 
+		//	 "purple": ["DoktorSAS"],
+		//	 "red" : ["fed"]
+		// }
+
+		void load_colors_data()
+		{
+			if (!utils::io::file_exists("colors/chat.json"))
+			{
+				printf("chat.json was not found\n");
+				return;
+			}
+
+			nlohmann::json obj;
+			try
+			{
+				obj = nlohmann::json::parse(utils::io::read_file("colors/chat.json").data());
+			}
+			catch (nlohmann::json::parse_error&)
+			{
+				printf("json parse error\n");
+				return;
+			}
+
+			static std::vector<std::string> keys = { "black", "red", "green", "yellow", "blue", "cyan", "purple", "white", "grey", "brown" };
+			static std::vector<std::string> values = { "^0", "^1", "^2", "^3", "^4", "^5", "^6", "^7", "^8", "^9" };
+
+			int index = 0;
+			for (const auto& entry_k : keys)
+			{
+				if (obj.contains( entry_k ))
+				{
+					std::vector<std::string> current = obj[entry_k];
+					for (const auto& entry_v : current)
+					{
+						colors_users[entry_v] = values.at(index).data();
+					}
+				}
+				index++;
+			}
+		}
+
+		const char* g_say_va_hook(const char* msg, char mode, char* team_color, char* name, 
```

Comment:
hook location is wrong in my opinion, after further inspection you only need to hook where the name is created and add the colour. There is no need to hook VA and it has never been done because it's usually a bad idea.

Reply:
Before making the hook on va I made other hooks. But none of them have any effect on chat. The player name does not contain the color and the caln tag does not contain it either. I haven't found any other working idea

Reply:
> hook location is wrong in my opinion, after further inspection you only need to hook where the name is created and add the colour. There is no need to hook VA and it has never been done because it's usually a bad idea.

i've always done this and had 0 problems 🤷‍♀️

Reply:
The right place to hook is CS_DisplayName where the name and clantag are built.

Reply:
> The right place to hook is CS_DisplayName where the name and clantag are built.

Refer to previous message. CS_DisplayName is called in G_Say, and the only way to hook it is hooking Plutonium and not the actual game.

Reply:
I would need to test that. I don't know what isn't called by the plutonium hook, what they use or don't use. Then again this isn't my pr so all of this should researched by the author before committing. Thanks for the useful info I didn't know that

Reply:
just for reference @diamante0018 - this is the beginning of Plutonium's hook on G_Say. It's very similar to BO2, it's just decompiled differently
![image](https://user-images.githubusercontent.com/64374267/156855489-2c661858-0e93-4afa-ae3f-120a5db337a9.png)

Reply:
Good research. I would then suggest to jump hook the game functionion in order to avoid problems with plutonium